lib/cell.rb

```diff
@@ -0,0 +1,63 @@
+class Cell
+
+  attr_reader :mine, :cleared, :flagged, :adjacent_mines
+
+  def initialize
+    @mine = false
+    @cleared = false
+    @flagged = false
+    @adjacent_mines = 0
+  end
+
+  # This ends the game!
+  def exploded?
+    mine && cleared
```

Comment:
Wow, such simple. The advantages of doing @cleared = false instead of @hidden = true